components/esp-tls/esp_tls_wolfssl.c

```diff
@@ -310,6 +313,12 @@ static esp_err_t set_client_config(const char *hostname, size_t hostlen, esp_tls
 #endif /* CONFIG_WOLFSSL_HAVE_ALPN */
     }
 
+#ifdef CONFIG_WOLFSSL_HAVE_OCSP
+    wolfSSL_CTX_EnableOCSPStapling((WOLFSSL_CTX *)tls->priv_ctx );
```

Comment:
Can you check the return codes for all the API?

Also, I think the order should be

`wolfSSL_CTX_EnableOCSP`
`wolfSSL_CTX_EnableOCSPStapling`
`wolfSSL_UseOCSPStapling`

Reply:
Please see:

* [wolfSSL_CTX_EnableOCSP](https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_ctx_enableocsp)
* [wolfSSL_CTX_EnableOCSPStapling](https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_ctx_enableocspstapling)
* [wolfSSL_UseOCSPStapling](https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_useocspstapling)

The web page appears to not always immediately auto-scroll to the correct location for me, so give it a few seconds, make sure the entire page loads. Ctrl-F might help navigate to the specific API documentation.

Reply:
Thanks Jim, I didn't realize there are separate manual pages (I was missing the doxygen docs I used to see in other places in wolfSSL;-). I quickly checked if there is additional useful information. The "wolfSSL_CTX_EnableOCSP" man page shows an example for a different function and is missing the "WOLFSSL_OCSP_CHECKALL" flag used in the example client. The others I think I got right from reading the source code!-)

Reply:
Glad to help. Having a robust integration with Espressif ESP-IDF is definitely on the roadmap for wolfSSL.

In the meantime, I've opened an issue regarding the documentation. See https://github.com/wolfSSL/wolfssl/issues/7564

Reply:
Hi @frankencode -

@embhorn has https://github.com/wolfSSL/wolfssl/pull/7567 with  an [update](https://github.com/wolfSSL/wolfssl/pull/7567/files) - would you please take a look?

Reply:
@gojimmypi Looks good to me. Btw. I figured I couldn't find the docs initially because that folder is not copied by your IDE/Espressif/*/setup.sh script. When I find the time I will update esp-wolfssl (or create something similar for my convenience).

Reply:
Hi @frankencode - cool, thanks for taking a look. The PR was merger earlier.

yes - the script currently only copies the essential code for build and not the docs. I'm curious why you choose the script copy method? The new `CMakeLists.txt` allows you to point the component at a GitHub repo directory. For me, it is much easier to see and track changes with the cmake file.

I'm always interested in how others are using the wolfSSL library. 

Cheers

Reply:
Jim, I have to look at your new method at some later time, again. It initially turned me off that you couldn't follow the common conventions there for the example CMakeLists.txt. Components are searched for by EXTRA_COMPONENT_DIRS or similar. Paths to some local installation directories do not belong into a CMakeLists file! Also, as I told you before, with the current setup there are missing Kconfig variables which allow to select wolfSSL as esp-tls backend. And I think when I did add those I ran into many build issues... I'm currently only testing/prototyping. I surely going to clean up my build setup later-on and keep you posted when I figured out how to build wolfSSL latest version cleanly for esp-idf. So far updating esp-wolfssl looks like the best bet to me.

Reply:
Thank for the feedback! You are absolutely right: the Kconfig definitely needs improvement. I'll be giving that attention soon. Let me know if there are any specifics that you think should be given priority. Perhaps you have a sample? Ideally, I'd of course like _all_ of the wolfSSL settings to be configurable via Kconfig... but there are many.

> Paths to some local installation directories do not belong into a CMakeLists file!

Although yes, there's flexibility to do this, the default behavior is to simply search the parent directories in the case of the [examples](https://github.com/wolfSSL/wolfssl/tree/master/IDE/Espressif/ESP-IDF/examples) and use the wolfSSL a few directories up. 

In the case of a different directory, such as in my case - I have this:

`workspace/wolfssl`
`workspace/myproject_using_wolfssl`
`workspace/myproject2_using_wolfssl`

The value being ensuring all projects are pointing to a fixed wolfSSL git source code repository, and easily point to an alternate as desired. If there's a `wolfssl` directory _in_ the `myproject`, then that one will be used.

I'll be sure and take a look at your `EXTRA_COMPONENT_DIRS` suggestion. I'm always open to making improvements.

I do believe that still works for a completely different location, as there's an Apple Homekit project doing that with wolfSSL.

Kind regards.